homeassistant/components/intent.py

```diff
@@ -0,0 +1,178 @@
+"""Module to coordinate user intentions."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+
+DOMAIN = 'intent'
+_LOGGER = logging.getLogger(__name__)
+
+SLOT_SCHEMA = vol.Schema({
+    vol.Optional('type'): str,
+})
+
+SPEECH_TYPE_PLAIN = 'plain'
+SPEECH_TYPE_SSML = 'ssml'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup intent."""
+    hass.intent = IntentRegistry(hass)
```

Comment:
So I see the `intent` component to be similar to the `http` component: it is part of the core but it is optional.

_Very small side note: I was also thinking if we should extract `service`, `config`, `eventbus`, `statemachine` too, but that's for another topic_